src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/ProjectImportsSubTreeProvider.cs

```diff
@@ -0,0 +1,280 @@
+    /// including props/targets from the SDK, targets, etc.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// To reduce clutter and memory usage, this tree is only visible (and populated) when
+    /// the "Show all Files" feature of Solution Explorer is enabled.
+    /// </para>
+    /// <para>
+    /// This feature can be enabled and disabled via the <c>ProjectImportsTree</c> project capability.
+    /// </para>
+    /// </remarks>
+    [Export(ExportContractNames.ProjectTreeProviders.PhysicalViewRootGraft, typeof(IProjectTreeProvider))]
+    [AppliesTo(Capability)]
+    internal sealed class ProjectImportsSubTreeProvider : ProjectTreeProviderBase, IProjectTreeProvider, IShowAllFilesProjectTreeProvider
+    {
+        public const string Capability = "ProjectImportsTree";
+
+        private static readonly ProjectImageMoniker s_rootIcon = KnownMonikers.ImportSettings.ToProjectSystemType();
+        private static readonly ProjectImageMoniker s_nodeIcon = KnownMonikers.TargetFile.ToProjectSystemType();
+
+        private static ProjectTreeFlags ProjectImportsTreeRootFlags { get; } = ProjectTreeFlags.Create(
+            ProjectTreeFlags.Common.BubbleUp |              // sort to top of tree, not alphabetically
+            ProjectTreeFlags.Common.VirtualFolder |
+            ProjectTreeFlags.Common.DisableAddItemFolder);
+        
+        private static ProjectTreeFlags ProjectImportFlags { get; } = ProjectTreeFlags.Create(
+            ProjectTreeFlags.Common.SourceFile |
+            ProjectTreeFlags.Common.NonMemberItem);         // enable double-click/enter to edit
+
+        private readonly IActiveConfiguredProjectSubscriptionService _projectSubscriptionService;
+        private readonly IUnconfiguredProjectTasksService _unconfiguredProjectTasksService;
+
+        private IDisposable? _subscription;
+        private bool _showAllFiles;
+
+        [ImportingConstructor]
+        internal ProjectImportsSubTreeProvider(
+            IProjectThreadingService threadingService,
+            IActiveConfiguredProjectSubscriptionService projectSubscriptionService,
+            IUnconfiguredProjectTasksService unconfiguredProjectTasksService,
+            UnconfiguredProject unconfiguredProject)
+            : base(threadingService, unconfiguredProject)
+        {
+            _projectSubscriptionService = projectSubscriptionService;
+            _unconfiguredProjectTasksService = unconfiguredProjectTasksService;
+        }
+
+        public bool ShowAllFiles
+        {
+            get => _showAllFiles;
+            set
+            {
+                // Ensure we no-op if the value didn't change, avoiding race conditions
+                lock (SyncObject)
+                {
+                    if (_showAllFiles == value)
+                    {
+                        return;
+                    }
+
+                    _showAllFiles = value;
+                }
+
```

Comment:
I dislike this pattern of putting these methods in a setter. Can we not do this?